File: Ruby/BetaPowerEfficientSpaceStation.rb
#encoding:utf-8

require_relative 'BetaPowerEfficientSpaceStationToUI'
require_relative 'Transformation'
require_relative 'PowerEfficientSpaceStation'

module Deepspace

    class BetaPowerEfficientSpaceStation < PowerEfficientSpaceStation

        @@EXTRAEFFICIENCY = 1.2

        def initialize(station)
            self.copy(station)
            @dice = Dice.new
        end

        def fire
            f = super
            if @dice.extraEfficiency
                f *= @@EXTRAEFFICIENCY
            end
            f 
        end



    end

end

File: Java/deepspace/src/deepspace/BetaPowerEfficientSpaceStation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author fjsae
 */
public class BetaPowerEfficientSpaceStation extends PowerEfficientSpaceStation{
    private static final float EXTRAEFFICIENCY = 1.2f;
    private Dice dice;
    
    public BetaPowerEfficientSpaceStation(SpaceStation station){
        super(station);   
    }
    
    @Override
    public float fire(){
        float f = super.fire();
        if(dice.extraEfficiency()){
            f *= EXTRAEFFICIENCY;
        }
        
        return f;
    }
}
